src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -145,4 +148,48 @@ public static ResourceTypeSchema load(final JSONObject schemaJson) {
     public Map<String, Object> getUnprocessedProperties() {
         return Collections.unmodifiableMap(this.unprocessedProperties);
     }
+
+    public boolean hasWriteOnlyProperties(final JSONObject resourceModel) {
+        return this.getWriteOnlyPropertiesAsStrings().stream()
+            .anyMatch(writeOnlyProperty -> hasProperty(RegExUtils.removeFirst(writeOnlyProperty, "/properties"), resourceModel));
+    }
+
+    public boolean hasProperty(final String property, final JSONObject resourceModel) {
+        return hasProperty(new JSONPointer(property), resourceModel);
+    }
+
+    public boolean hasProperty(final JSONPointer property, final JSONObject resourceModel) {
+        try {
+            return property.queryFrom(resourceModel) != null;
+        } catch (JSONPointerException e) {
+            return false;
+        }
+    }
+
+    public void removeWriteOnlyProperties(final JSONObject resourceModel) {
+        this.getWriteOnlyPropertiesAsStrings().stream()
+            .forEach(writeOnlyProp -> removeProperty(new JSONPointer(RegExUtils.removeFirst(writeOnlyProp, "/properties")),
+                resourceModel));
+    }
+
+    public static void removeProperty(final JSONPointer property, final JSONObject resourceModel) {
+        String refParts = property.toString();
+        // split pointer into tokens, removing first delimiter
+        List<String> refTokens = Arrays.asList(RegExUtils.removeFirst(refParts, "/").split("/"));
+        final String key = refTokens.get(refTokens.size() - 1);
+        try {
+            // if size is more than one, fetch parent object/array of key to remove so that
```

Comment:
Thinking maybe that rather than remove the property and give the appearance that it's not set at all, we redact/mask the value instead - this would indicate clearly that the property is set but not what its value is. Thoughts?

Reply:
Yeah that makes sense, but might get tricky if the fields have patterns/min/max length constraints, though I am not sure if that would be a common use case. The model after being used could potentially not be valid against the schema, though the same thing may be true from removing a property.

Reply:
I can see two cases:

1. Redacting the field with a new or blank value could fail property validations.
2. Removing the filed entirely could fail validation if the writeOnly property is required.

One option could be to enforce that writeOnly properties must not also be required properties. If writeOnly properties are allowed to be required, then that means all reads of the schema are not valid representations of the resource. What do you think?

Reply:
A model with redacted fields failing validation might be an overall positive side-effect - it would guarantee that the model could not be accidentally applied (for example to correct drift).